src/sys/wait.rs

```diff
@@ -1,58 +1,100 @@
-use libc::{self, c_int};
+//! This module contains the `wait()` and `waitpid()` functions, which are used to wait on and
+//! obtain status information from child processes. These provide more granular control over
+//! child management than the primitives provided by Rust's standard library, and are critical
+//! in the creation of shells and managing jobs on *nix platforms.
+//!
+//! # Example
+//!
+//! ```rust
+//! use nix::sys::wait::*;
+//! use nix::unistd::Pid;
+//!
+//! let pid = Pid::from_raw(17563);
+//! loop {
+//!     match waitpid(PidGroup::ProcessGroupID(pid), WUNTRACED) {
+//!         Ok(WaitStatus::Exited(pid, status)) => {
+//!             println!("Process '{}' exited with status '{}'", pid, status);
+//!             break
+//!         },
+//!         Ok(WaitStatus::Stopped(pid, signal)) => {
+//!             println!("Process '{}' stopped with signal '{:?}'", pid, signal);
+//!         },
+//!         Ok(WaitStatus::Continued(pid)) => {
+//!             println!("Process '{}' continued", pid);
+//!         },
+//!         Ok(_) => (),
+//!         Err(why) => {
+//!             println!("waitpid returned an error code: {}", why);
+//!             break
+//!         }
+//!     }
+//! }
+//! ```
+
+use libc::{self, c_int, pid_t};
 use {Errno, Result};
 use unistd::Pid;
 
 use sys::signal::Signal;
 
-mod ffi {
-    use libc::{pid_t, c_int};
-
-    extern {
-        pub fn waitpid(pid: pid_t, status: *mut c_int, options: c_int) -> pid_t;
-    }
-}
-
-#[cfg(not(any(target_os = "linux",
-              target_os = "android")))]
-libc_bitflags!(
-    pub flags WaitPidFlag: c_int {
-        WNOHANG,
-        WUNTRACED,
-    }
-);
-
-#[cfg(any(target_os = "linux",
-          target_os = "android"))]
 libc_bitflags!(
+    /// Defines optional flags for the `waitpid` function.
     pub flags WaitPidFlag: c_int {
+        /// Do not suspend execution of the calling thread if the status is not immediately
+        /// available for one of the child processes specified by pid.
         WNOHANG,
+        /// The status of any child processes specified by pid that are stopped, and whose status
+        /// has not yet been reported since they stopped, shall also be reported to the requesting
+        /// process
         WUNTRACED,
+        /// Waits for children that have terminated.
+        #[cfg(any(target_os = "android", target_os = "linux"))]
         WEXITED,
+        /// Report the status of any continued child process specified by pid whose status has not
+        /// been reported since it continued from a job control stop.
+        #[cfg(any(target_os = "android", target_os = "linux"))]
         WCONTINUED,
```

Comment:
WCONTINUED is available pretty much everywhere. Definitely on `freebsd`, `openbsd`, `bitrig`, `dragonfly`, `netbsd` and `solaris`.